### PR TITLE
Recipe App

### PR DESCRIPTION
**Pull request by Victoria129**
### In this PR, we have implemented the following:

- [ ] Login page and registration page:
- [ ] Should be built with Devise.
- [ ] Food list 
- [ ] Should display a list of food added by the logged-in user as in the wireframe (display also quantity of a given food).
- [ ] Should lead to a form that allows users to add new food.
- [ ] Recipes list:
- [ ] Should display a list of recipes created by the logged-in user as in the wireframe.
- [ ] Should lead to recipe details.
- [ ] If the user is the owner of the recipe, should allow the user to delete it.
- [ ] Public recipe list:
- [ ] Should display a list of all public recipes ordered by newest as in the wireframe.
- [ ] Should lead to recipe details.
- [ ] Recipe details:
- [ ] Should display a toggle button that allows for a recipe to be made public or private.
- [ ] If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
- [ ] If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
- [ ] General shopping list view 
- [ ] Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
- [ ] Should count the total food items and total price of the missing food.
- [ ] Make sure there are no N+1 queries happening.
- [ ] Create a navigation menu that allows users to open all of the pages you created.
- [ ] Write unit and integration tests

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<div class="container mt-5">
+  <div class="row justify-content-center">
+    <div class="col-md-6">
+      <div class="card">
+        <div class="card-header"><h2 class="mb-0">Log in</h2></div>
+        <div class="card-body">
+          <%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
+            <div class="form-group">
+              <%= f.label :email %>
+              <%= f.email_field :email, autofocus: true, autocomplete: "email", class: "form-control" %>
```
Comment (VICTOR-CHUKWU) [inline app/views/devise/sessions/new.html.erb:10]: - [ ] an error occurred running this app about an undefined email, kindly fix it
image

File: spec/helpers/shopping_lists_helper_spec.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe ShoppingListHelper, type: :helper do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (VICTOR-CHUKWU) [inline spec/helpers/shopping_lists_helper_spec.rb]: - [ ] There is an error stopping your test from running, kindly fix it
image

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @Victoria129  and @Jawad-Nazari  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/sessions/new.html.erb (hunk 1 above)
Comment (Jawad-Nazari) [reply app/views/devise/sessions/new.html.erb:10]: Hi @VICTOR-CHUKWU , 
Thanks for your time and for reviewing our code. We will come up with a solution. 🙏

File: spec/helpers/shopping_lists_helper_spec.rb (hunk 2 above)
Comment (Jawad-Nazari) [reply spec/helpers/shopping_lists_helper_spec.rb]: Hi @VICTOR-CHUKWU ,

 Thanks for your advice. All the tests are passed now.
image

File: app/views/devise/sessions/new.html.erb (hunk 1 above)
Comment (zieeco) [reply app/views/devise/sessions/new.html.erb:10]: - As pointed out by the previous reviewer, I still get the same error when running your app. Kindly fix it.



Screenshot from 2023-08-12 14-25-38

File: spec/helpers/shopping_lists_helper_spec.rb (hunk 2 above)
Comment (zieeco) [reply spec/helpers/shopping_lists_helper_spec.rb]: - Also, your tests are still not passing. kindly fix this.

image

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @Team,





Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the changes, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice design. ✔️
- Descriptive PR. ✔️
- Nice implementing the required changes :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions


- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._**


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/recipes_spec.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe 'Recipes', type: :request do
+  describe 'GET /index' do
+    it 'returns http success' do
+      get '/recipes/index'
+      expect(response).to have_http_status(:success)
+    end
+  end
```
Comment (codepantha) [inline spec/requests/recipes_spec.rb]: - [ ] This test is failing as you can see below 👇. Please try to make it pass 🙏 
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi team 👋, 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



## Highlights 🎯 
✔️ Descriptive PR
✔️ Added tests with RSpec and Capybara 📌 
✔️ No linter errors 🟢 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi team 👋, 

Great job implementing the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights 🎯 
✔️ Descriptive PR
✔️ Added tests with RSpec and Capybara 📌
✔️ All tests are now passing 📌 
✔️ No linter errors 🟢

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**